addons/mail/models/mail_channel.py

```diff
@@ -1096,6 +1097,10 @@ def channel_fetch_slot(self):
         direct_message_channels = self.search([('channel_type', '=', 'chat'), ('id', 'in', pinned_channels.ids)])
         values['channel_direct_message'] = direct_message_channels.channel_info()
 
+        # get the pinned 'group chat' channel
+        group_chat_channels = self.search([('channel_type', '=', 'group'), ('id', 'in', pinned_channels.ids)])
+        values['channel_group_chat'] = group_chat_channels.channel_info()
```

Comment:
Note (for another PR?) it's useless to return each channel type as a different key. init messaging could be simplified to simply return all pinned channels of the current user, without distinction.